### PR TITLE
Fix/refactor out sugardb

### PR DESCRIPTION
**Pull request by mrsasha**

### CODE REVIEW

File: android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ * Created by skraynick on 16-03-14.
+ */
+@EqualsAndHashCode
+public class BeaconAction {
+
+    public static final String SHARED_PREFS_TAG = "BeaconActions";
+
+    public static final long NO_DATE = Long.MIN_VALUE;
+
+    @Expose
```
Comment (skraynick) [inline android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java]: Not sure we need the `@Expose` anymore. They were needed for Sugar... If I remember correctly.

File: android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    @Getter
+    @Setter
+    private boolean keepForever;
+
+    /**
+     * Default constructor as required by SugarORM.
+     */
+    public BeaconAction() {
```
Comment (skraynick) [inline android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java]: Delete, default was needed for sugar.

File: android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconScan.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+ * @version 16-03-14
+ */
+@EqualsAndHashCode
+public class BeaconScan {
+
+    public static final String SHARED_PREFS_TAG = "BeaconScans";
+
+    public static final long NO_DATE = Long.MIN_VALUE;
+
+    @Expose
```
Comment (skraynick) [inline android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconScan.java]: Same as above.

File: android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconScan.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    @Getter
+    @Setter
+    private long createdAt;
+
+    /**
+     * Default constructor as required by SugarORM.
+     */
+    public BeaconScan() { //TODO do we need it?
```
Comment (skraynick) [inline android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconScan.java]: nope. Delete.

File: android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconScan.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        return isEntry() ? ScanEventType.ENTRY.getMask() : ScanEventType.EXIT.getMask();
+    }
+
+    public String getPid() {
+        return this.getProximityUUID().replace("-", "") + String.format("%1$05d%2$05d", this.getProximityMajor(), this.getProximityMinor());
+    }
+
+    /**
+     * Creates a SugarScan Object.
+
```
Comment (skraynick) [inline android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconScan.java]: No space

**Comment by skraynick** [comment]
Just the few things @mrsasha ..

File: android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java (hunk 1 above)
Comment (mrsasha) [inline android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java]: we need it for GSON serialization, it's setup to expose only annotated fields.

File: android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java (hunk 2 above)
Comment (mrsasha) [inline android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java]: I removed the comments, but apparently it's also useful to have it for Gson: http://stackoverflow.com/questions/18645050/is-default-no-args-constructor-mandatory-for-gson

File: android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java (hunk 1 above)
Comment (skraynick) [inline android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java]: okay cool..

File: android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    @Expose
+    @Getter
+    @Setter
+    private boolean keepForever;
+
+    public BeaconAction() {
+    }
+
+    /**
+     * Gets SugarAction object.
```
Comment (skraynick) [inline android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java]: Beacon Action?

File: android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    @Getter
+    @Setter
+    private String pid;
+
+    @Expose
+    @Getter
+    @Setter
+    private boolean keepForever;
+
+    public BeaconAction() {
```
Comment (skraynick) [inline android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java]: Don't think we need these defaults. We needed them for sugar.

Comment (mrsasha) [inline android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java]: also needed for Gson.

File: android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java (hunk 6 above)
Comment (mrsasha) [inline android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java]: ah, right.

File: android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java (hunk 7 above)
Comment (skraynick) [inline android-sdk/src/main/java/com/sensorberg/sdk/model/persistence/BeaconAction.java]: okay.. cool. will wait until checks are done and  Will merge.

**Pull request merged**